lesson16/main.js

```diff
@@ -0,0 +1,117 @@
+"use strict";
+const body = document.getElementById("body");
+let categoryTab = document.getElementById("js-categoryTab");
+const jsonURL = "http://myjson.dit.upm.es/api/bins/du3p";
+// const jsonURL = "https://myjson.dit.upm.es/api/bins/gdod";　データが空のjson
+async function fetchData(url) {
+   const response = await fetch(url);
+   const json = await response.json();
+   return json.data;
+}
+const createElementWithClassName = (element, name) => {
+   const createElement = document.createElement(element);
+   createElement.classList.add(name);
+   return createElement;
+};
+const displayMessage = (message) => {
+   const div = document.createElement("div");
+   body.appendChild(div);
+   div.textContent = message;
+}
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.id = "loadingIcon";
+   loadingIcon.src = "img/loading-circle.gif";
+   body.appendChild(loadingIcon);
+};
+const endLoading = () => {
+   const loadingIcon = document.getElementById("loadingIcon");
+   loadingIcon.remove();
+}
+
+createTabContents();
+async function createTabContents() {
+   startLoading();
+   let responseData;
+   try {
+      responseData = await fetchData(jsonURL);
+      if(responseData.length === 0) {
+         displayMessage("データはまだありません");
+         return;
+         } 
+   } catch (error) {
+      displayMessage(error);
+      console.log(error);
+   } finally {
+      endLoading();
+   }
+   createCategoryTab(responseData);
+   renderTabContainer(responseData);
+}
+const createCategoryTab = (values) => {
+   const fragment = document.createDocumentFragment();
+   for (let i = 0; i < values.length; i++) {
+      const categoryTabItem = createElementWithClassName("li", "category-tab__item")
+      categoryTabItem.id = `js-categoryTabItem${i}`;
+      const categoryTabText = createElementWithClassName("a", "category-tab__text");
+      categoryTabText.textContent = values[i].category
+      fragment.appendChild(categoryTabItem).appendChild(categoryTabText);
+   }
+   categoryTab.appendChild(fragment);
+   //初期表示のタブの選択はデータとして持っているがまだ理解できていないので、現段階ではNewsのタブとしてある
+   const newsTabItem = document.getElementById("js-categoryTabItem0");
+   newsTabItem.classList.add("is-active");
+}
+
+const renderTabContainer = (values) => {
+   const tabFragment = document.createDocumentFragment();
+   const tabContainer = createElementWithClassName("div", "tab-container");
+   tabContainer.id = "js-tabContainer";
+   for (let i = 0; i < values.length; i++) {
+      const tabContents = createElementWithClassName("div", "tab-contents");
+      tabContents.id = `js-tabContents${i}`;
+      const tabLists = createElementWithClassName("ul", "tab-contents__lists");
+      tabLists.id = `js-tabLists${i}`;
+      const tabImage = createElementWithClassName("img", "tab-contents__image");
+      tabImage.id = `js-tabImage${i}`;
```

Comment:
nits
I think you should take the id names that you are not using.

```suggestion
      const tabImage = createElementWithClassName("img", "tab-contents__image");
```

Reply:
Thank you for pointing this out.
I removed unnecessary code giving ID. 
I'm sorry, I'm missing some details to check.
[e9a79d9](https://github.com/taka0209/js_lesson/pull/18/commits/e9a79d9af321d28e9fc05e75c5b06a38456552db)